hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPServiceActor.java

```diff
@@ -1378,7 +1378,7 @@ public long monotonicNow() {
     private final BlockingQueue<Runnable> queue;
 
     CommandProcessingThread(BPServiceActor actor) {
-      super("Command processor");
+      setName("Command processor-" + getId());
```

Comment:
Why not using super()?

Reply:
Because the thread id is set during construction and not available to us yet.

Reply:
Thanks @madrob for your works. I don't think `getId()` will get additional information.
IMO, it may be better to change like the following?
`  super("Command processor for " + nnAddr);`

Reply:
Thanks for the suggestion, but I don't think that would be an improvement. Let me explain the motivation in more detail?

The id is just the numeric Java thread id, and it's enough to differentiate the command processors between each other when there are multiple DN running in the same process like in MiniDFSCluster during unit tests.

Putting the NN address in would not disambiguate the logs because they would all be for the same NN still. It would give more information, sure, but not actually helpful information.

With my change, the log messages would have (Command processor-56) or -68 or whatever the thread was. Again, just enough to differentiate them from one another, which is what I needed for tracing their lifecycle and operation.

If there's a DN address we can use in the thread name instead, then that's good too but I don't know enough about Hadoop internals to find that.

Reply:
> Putting the NN address in would not disambiguate the logs because they would all be for the same NN still. It would give more information, sure, but not actually helpful information.

Sorry I don't this information. IIUC, now each command processor match to block pool one by one. and actually `nnAddr` includes hostname/port together. I mean that it could different each other even for MiniDFSCluster  framework. Right? for another way, with nnAddr, it could be helpful to dig when this thread meet issues.
Anyway, I don't disagree to add getId() also here. We could add both them to the thread name.

Reply:
I think the issue here is in case of MiniDfsCluster all datanodes log at one place.Means if we spin a MiniDfsCluster with 9 datanodes, all Command processor threads will log at same place, and we can't distinguish, which thread belongs to which datanode.
DN1 will also have same name & DN2 till DN9. If we add namenode address, then also the names of the thread will stay same right? All datanodes in a single MiniDfs will be connected to same set of namenodes, right?

May be adding DN address should be a good idea?

Reply:
Yes, exactly, @ayushtkn. All of the DN will have the same NN and then they will log the same thread name string.

For adding DN address, I'm not sure which fields in DN are considered stable for this purpose. Maybe `setName("Command processor for " + dn.getDisplayName());` will work?

Reply:
> I think the issue here is in case of MiniDfsCluster all datanodes log at one place.Means if we spin a MiniDfsCluster with 9 datanodes, all Command processor threads will log at same place, and we can't distinguish, which thread belongs to which datanode. DN1 will also have same name & DN2 till DN9. If we add namenode address, then also the names of the thread will stay same right? All datanodes in a single MiniDfs will be connected to same set of namenodes, right?
> 
> May be adding DN address should be a good idea?

Thanks @madrob @ayushtkn for your discussions and information. From my side, I am more concerned how to differ these threads service to which namespace when setup Federation production cluster. This may be helpful for digging issues when meet something not expected. I total agree to add `dn.getDisplayName()` or `getId()` or something else also that can improve the MiniDFSCluster log readable. IMO we should consider production env meanwhile.  FYI. Of course this is not the FATAL issue, my comment is not blocker. Please feel free to go ahead if possible.

Reply:
Makes sense, we should handle federation setup as well. That is what will help in actual prod env.